### PR TITLE
Add pre-commit hooks

### PR DESCRIPTION
**Pull request by ESKYoung**
Add pre-commit hooks, as implemented by @karlbaker02 in [cookiecutter-data-science-gds](https://github.com/ukgovdatascience/cookiecutter-data-science-gds), with some minor modifications for the new repository structure.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,6 +60,76 @@ cookiecutter https://github.com/ukgovdatascience/govcookiecutter
...
+pre-commit hook might be used to run any code linting automatically, providing any warnings before code is committed,
+ensuring that all of our code adheres to a certain quality standard.
+
+For this repo, we are using `pre-commit` for a number of purposes:
+- Checking for any secrets being committed accidentally;
+- Checking for any large files (over 5MB) being committed; and
+- Cleaning Jupyter notebooks, which means removing all outputs and execution counts.
+
+We have configured `pre-commit` to run automatically on _every commit_. By running on each commit, we ensure
+that `pre-commit` will be able to detect all contraventions and keep our repo in a healthy state.
```
Comment (karlbaker02) [inline README.md:76]: Did you mean to add the docs about `pre-commit` to the main repo README, or only the projects which this repo is here to create?

**Review by karlbaker02 — APPROVED** [review]
Looks good, one small comment to check intentions but happy to approve

File: README.md (hunk 1 above)
Comment (ESKYoung) [reply README.md:76]: Both actually - the main repo for anyone developing the cookiecutter, and the resultant repos that this cookiecutter creates as well!

**Review by karlbaker02 — APPROVED** [review]

**Comment by ESKYoung** [comment]
Great thanks @karlbaker02! Merging in now!

**Pull request merged**